src/App.vue

```diff
@@ -90,26 +110,129 @@ export default class App extends Mixins(DateMixin, FilingTemplateMixin, LegalApi
   @Action setCurrentDate!: ActionBindingIF
   @Action setCertifyStatementResource!: ActionBindingIF
   @Action setNameRequestState!: ActionBindingIF
+  @Action setAuthRoles: ActionBindingIF
 
   // Local Properties
   private filingData: Array<FilingDataIF> = []
   private draftFiling: IncorporationFilingIF
+  private isDraft: boolean = false
+  private accountAuthorizationErrorDialog: boolean = false
+  private nameRequestInvalidErrorDialog: boolean = false
+  private nameRequestInvalidType: string = ''
+  private evaluatingPreconditions: boolean = true
 
   private async created (): Promise<any> {
-    // Mock the nrNumber and Data:
-    this.setNameRequestState({ nrNumber: 'NR7654560', entityType: 'BC' })
+    // Evaluate name request pre conditions
+    const nameRequest = await this.evaluateNRPreconditions()
+    if (nameRequest && nameRequest.nrNum && nameRequest.isConsumable) {
+      // TODO: Handling different NR Formats
+      // Will probably change once we proxy through legal API with a consistent format
+      this.setNameRequestState(
+        { nrNumber: nameRequest.nrNum.replace('NR ', ''),
+          entityType: nameRequest.requestTypeCd,
+          filingId: null })
+      this.setCurrentDate(this.dateToUsableString(new Date()))
 
-    this.setCurrentDate(this.dateToUsableString(new Date()))
+      try {
+        // Retrieve draft filing if it exists for the nrNumber specified
+        this.draftFiling = await this.fetchDraft()
 
-    try {
-      // Retrieve draft filing if it exists for the nrNumber specified
-      this.draftFiling = await this.fetchDraft()
+        // Parse the draft data into the store if it exists
+        this.draftFiling && this.parseDraft(this.draftFiling)
 
-      // Parse the draft data into the store if it exists
-      this.draftFiling && this.parseDraft(this.draftFiling)
-    } catch (e) {
-      // TODO: Catch a flag from the api, if there is an error to be handled.
+        // Inform the router view we are resuming a draft and to update ui
+        this.isDraft = true
+      } catch (e) {
+        // TODO: Catch a flag from the api, if there is an error to be handled.
+      }
     }
+    this.evaluatingPreconditions = false
+  }
+
+  /**
+   * Redirect to business URL
+   */
+  private redirectToBusinessUrl () : void {
+    const businessUrl: string = sessionStorage.getItem('BUSINESSES_URL') || ''
+    window.location.assign(businessUrl)
+  }
+
+  private authAPIURL () {
+    return sessionStorage.getItem('AUTH_API_URL')
+  }
+  /**
+   * Fetch Authorizations by NR number
+   */
+  private getNRAuthorizations (nrNumber) {
+    const url = nrNumber + '/authorizations'
+    const config = {
+      baseURL: this.authAPIURL() + 'entities/'
+    }
+    return axios.get(url, config)
```

Comment:
Might be returning a Promise here? could look something like `private getNRAuthorizations (nrNumber): Promise<any>`

Reply:
Updated return and parameter type